tests/test_middleware_interface.py

```diff
@@ -701,11 +773,27 @@ def test_run_pipeline_cascading_exception(self):
              unittest.mock.patch("activator.middleware_interface.SeparablePipelineExecutor.run_pipeline") \
                 as mock_run, \
              unittest.mock.patch("lsst.dax.apdb.ApdbSql.containsVisitDetector") as mock_query:
-            mock_run.side_effect = RuntimeError("The pipeline doesn't like you.")
+            mock_run.side_effect = ValueError("Error: not computable")
             mock_query.side_effect = psycopg2.OperationalError("Database? What database?")
             with self.assertRaises(NonRetriableError):
                 self.interface.run_pipeline({1})
 
+    def test_run_pipeline_early_exception_novisit(self):
+        """Test behavior when execution fails in ISR-only processing.
+        """
+        self._prepare_run_pipeline()
+
+        with unittest.mock.patch(
+                "activator.middleware_interface.SeparablePipelineExecutor.pre_execute_qgraph"), \
+             unittest.mock.patch("activator.middleware_interface.SeparablePipelineExecutor.run_pipeline") \
+                as mock_run, \
+             unittest.mock.patch.object(self.interface, "main_pipelines", pipelines_minimal), \
+             unittest.mock.patch("lsst.dax.apdb.ApdbSql.containsVisitDetector") as mock_query:
+            mock_run.side_effect = ValueError("Error: not computable")
+            mock_query.return_value = False
+            with self.assertRaises(PipelineExecutionError):
+                self.interface.run_pipeline({42})
```

Comment:
A comment here reminding/explaining that group 42 was constructed as engineering-only data would be helpful.